### PR TITLE
Use execute ipa commands

### PR DESCRIPTION
**Pull request by t-woerner**
execute_ipa_commands replces the check mode exit, the loop over the
generated commands and also in the member failure handling for modules
with member support.

### CODE REVIEW

**Review by rjeffman — COMMENTED** [review]
This is a nice reduction of duplicate code!

There are two test errors (group and hostgroup with membermanager), that should be fixed.

Also, I see that there are two versions of handle_result that are used multiple times in different modules. What about having those versions in ansible_module_freeipa for reuse by the modules?

**Comment by t-woerner** [comment]
The question is if an error found in handle_result should lead to fail_json immediately or after other commands have been executed. Right now we have modules that fail immediately and others that fail later.

**Review by rjeffman — APPROVED** [review]
lgtm.

**Comment by rjeffman** [comment]
@t-woerner I believe these changes in the modules to reduce `handle_result()` to a common code (or two versions of it, one for modules with members, one for modules without), can be done later, in a new round of refactoring.

File: plugins/modules/ipadnsrecord.py [hunk 1]
```diff
@@ -1408,6 +1408,13 @@ def define_commands_for_absent_state(module, zone_name, entry, res_find):
     return _commands
 
 
+# pylint: disable=unused-argument
+def exception_handler(module, ex):
+    if ex in [ipalib_errors.EmptyModlist, ipalib_errors.DuplicateEntry]:
```
Comment (rjeffman) [inline plugins/modules/ipadnsrecord.py:1413]: Would it be better to test for the instance type? Like `if isinstance(ex, (ipalib_errors.EmptyModlist, ipalib_errors.DuplicateEntry])):`.

Comment (t-woerner) [reply plugins/modules/ipadnsrecord.py]: Fixed

**Review by rjeffman — APPROVED** [review]
lgtm.

**Pull request merged**